Add file upload functionality and AWS S3 integration

gurkirtheerey opened the pull request:
- Implemented a new API route for file uploads using AWS S3 in `app/api/upload/route.ts`.
- Enhanced `CreateRecipeModal` to support image uploads, including file handling and integration with the new API.
- Updated `createRecipeSchema` to require an image file for recipe creation.
- Added necessary AWS SDK dependencies in `package.json` and `package-lock.json` for S3 operations.
- Introduced utility functions for S3 file uploads in `lib/utils/s3.ts`.

gurkirtheerey committed:
Add file upload functionality and AWS S3 integration

- Implemented a new API route for file uploads using AWS S3 in `app/api/upload/route.ts`.
- Enhanced `CreateRecipeModal` to support image uploads, including file handling and integration with the new API.
- Updated `createRecipeSchema` to require an image file for recipe creation.
- Added necessary AWS SDK dependencies in `package.json` and `package-lock.json` for S3 operations.
- Introduced utility functions for S3 file uploads in `lib/utils/s3.ts`.

sunshinemoua commented:
![Screenshot 2025-04-04 at 11 08 33 AM](https://github.com/user-attachments/assets/8c373fbe-4361-494f-b251-ed703ef58c6e)
Form is stuck here.

sunshinemoua commented on lib/schemas/recipe.ts:
```diff
@@ -5,6 +5,7 @@ export const createRecipeSchema = z.object({
   description: z.string().min(1, { message: 'Description is required.' }),
   ingredients: z.string().min(1, { message: 'Ingredients are required.' }),
   instructions: z.string().min(1, { message: 'Instructions are required.' }),
+  image: z.instanceof(File, { message: 'Image is required.' }),
```
I don't think it's necessary for the image to be required. It should be optional and they can add it later.

sunshinemoua commented on app/api/upload/route.ts:
Add comments please!

sunshinemoua commented on lib/utils/s3.ts:
Add comments please!!

gurkirtheerey committed:
Enhance image upload handling in CreateRecipeModal

- Added validation for file size and type in the image upload process.
- Updated handleUpload function to handle errors and provide user feedback via toast notifications.
- Made the image field optional in the createRecipeSchema to allow recipe creation without an image.
- Reorganized the file input section in the modal for improved user experience.

gurkirtheerey commented:
> ![Screenshot 2025-04-04 at 11 08 33 AM](https://private-user-images.githubusercontent.com/52899934/430495394-8c373fbe-4361-494f-b251-ed703ef58c6e.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDM3OTE2NzQsIm5iZiI6MTc0Mzc5MTM3NCwicGF0aCI6Ii81Mjg5OTkzNC80MzA0OTUzOTQtOGMzNzNmYmUtNDM2MS00OTRmLWIyNTEtZWQ3MDNlZjU4YzZlLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTA0MDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwNDA0VDE4MjkzNFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTgzYmEzOGI5MWE0MjRlM2E1MDI1ZDc2OTY4ZGUyOGQ2MzkyN2JjNTE2YjkyYWU5OWUyMjhkNDQ3OGUyYjVhMGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.AV4adDgSggDXflLnvzKq9yc4Uyqwp7zwh1WtWiozxp4) Form is stuck here.

fixed

gurkirtheerey committed:
Enhance file upload API and S3 utility functions

- Added detailed documentation comments to the file upload API in `app/api/upload/route.ts` for better understanding of functionality.
- Updated the S3 utility function in `lib/utils/s3.ts` with comments explaining the upload process and the return value of the public URL.

gurkirtheerey committed:
Remove image upload error handling from CreateRecipeModal
- This change allows users to create recipes without requiring an image upload

gurkirtheerey committed:
Refactor recipe image handling and enhance sidebar functionality

- Replaced SidebarTrigger with AppSidebarTrigger for better mobile sidebar management.
- Added new recipe error types for improved file upload validation in CreateRecipeModal.
- Updated image rendering logic in RecipeCard and RecipePage to handle cases without images, displaying a placeholder icon.
- Enhanced error handling in CreateRecipeModal to provide user feedback for file upload issues.

sunshinemoua commented on app/recipes/[id]/page.tsx:
Add edit image functionality next

sunshinemoua reviewed (approved):
Good job!!
